app/browser/api/ledger.js

```diff
@@ -982,13 +983,21 @@ const pageDataChanged = (state, viewData = {}, keepInfo = false) => {
 
   let info = pageDataState.getLastInfo(state)
   const tabId = viewData.tabId || pageDataState.getLastActiveTabId(state)
-  const location = viewData.location || locationDefault
+  let location = viewData.location || locationDefault
 
   if (!synopsis) {
     state = addNewLocation(state, locationDefault, tabId)
     return state
   }
 
+  if (UrlUtil.isUrlPDF(location)) {
+    location = UrlUtil.getUrlFromPDFUrl(location)
```

Comment:
it should be sufficient to just use `getLocationIfPDF` here since that is the function used to get the URL shown in the urlbar when a pdf.js URL is loaded

Reply:
that function is also used for bookmarks, history, etc.

Reply:
problem with that function is that it allows files as well, where we need to only check/have url's. This is the reason why I crated new function

Reply:
@NejcZdovc `getViewerUrl` is actually used by PDFJS for all URLs, not just file URLs - see https://github.com/brave/browser-laptop/blob/master/app/pdfJS.js#L101. IIRC the viewer URL is then redirected to the `chrome-extension://{pdfjs_id}/{url}` URL. however, if the redirect doesn't happen, we still want to display the real URL, not the viewer URL. that is why `getLocationIfPDF` also handles the viewer URL.

Reply:
👍